### ISSUE TITLE
Modal-related events not firing

### ISSUE THREAD

**Issue by LeGast00n** [twbs/bootstrap#19872]
Hi there,

Running on several Bootstrap versions (I tried 3.1.1 and now I tried the latest), I've been encountering a bug with modal-related events (show, shown, etc.).
It happens on Safari and Chrome (not tested on other browsers). Strangely I have checked my code in JSFiddle and it works.
Now, going into debugging and setting a breakpoint just before the event is fired, I could confirm that:
- firing manually and directly the event onto my object works -  $("#myModal").trigger("shown.bs.modal")
- firing manually the event using Bootstrap's piece of code does not work - this.$element.trigger(e)
- wrapping the object as a JQuery object works - $(this.$element).trigger(e)

Please note that at one point, when troubleshooting using the console, I managed to get the event firing work... only once.

So I guess it's probably a very annoying browser-related issue.

All in all, I'd recommend the JQuery wrapping of the object before firing the event.

cheers
LA

**Comment by cvrebert** [comment]
> Strangely I have checked my code in JSFiddle and it works.

This generally suggests that the problem lies in your specific webpage rather than in Bootstrap itself.
For troubleshooting help, try StackOverflow, IRC, or Slack. See https://github.com/twbs/bootstrap/blob/master/README.md#community for links.
If you can craft a reproducible reduced test case for the problem, feel free to open a new issue with the test case and we'll investigate.
